### PR TITLE
[NNC] Add Softplus operator

### PR DESCRIPTION
**Pull request by anijain2305**
Summary: Adding softplus operator lowering for NNC. Enabling element wise fusion as well.

Test Plan: Added a test in test_jit_fuser.py

Differential Revision: D30736449

### CODE REVIEW

File: test/test_jit_fuser_te.py [hunk 1]
```diff
@@ -1255,6 +1255,7 @@ def apply(fn):
             F.hardtanh,
             F.hardsigmoid,
             F.hardswish,
+            F.softplus,
```
Comment (bertmaher) [inline test/test_jit_fuser_te.py:1258]: Based on test failures, it looks like this should also be added to `works_list` lower in this file

**Review by bertmaher — CHANGES_REQUESTED** [review]
Other than the failing test this looks good, please update the test and I'll happily approve!

**Comment by bertmaher** [comment]
It would be cool to run these benchmarks and see if the fused `mish` is now on par with the native implementation: https://benjaminwarner.dev/2021/07/19/benchmarking-pytorch-native-mish

File: test/test_jit_fuser_te.py (hunk 1 above)
Comment (anijain2305) [reply test/test_jit_fuser_te.py:1258]: Added later.

**Review by bertmaher — APPROVED** [review]
Nice work, thanks!

**Pull request closed without merging**